app/util/concurrent/BackgroundProcessor.scala

```diff
@@ -36,19 +35,28 @@ object BackgroundProcessor {
     )
   }
 
-  type SendType[T] = Tuple2[Option[Throwable], Option[T]]
-  def send[PROC_RES,RESPONSE](cmd: BackgroundProcess[PROC_RES])(result: SendType[PROC_RES] => RESPONSE)(implicit mf: Manifest[PROC_RES]) = {
-    val p = Promise[RESPONSE]()
-    
-    ask(ref, cmd)(cmd.timeout).mapTo[PROC_RES].onComplete {
-      case Success(v: PROC_RES) => p success result(Tuple2(None, Some(v)))
-      case Failure(e) => e match {
-        case t: TimeoutException =>
-          p success result(Tuple2(Some(SexyTimeoutException(cmd.timeout)), None))
-        case _ =>
-          p success result(Tuple2(Some(e), None))
-      }
+  type SendType[T] = (Option[Throwable], Option[T])
+
+  def send[PROC_RES,RESPONSE](cmd: BackgroundProcess[PROC_RES])(result: SendType[PROC_RES] => RESPONSE)(implicit mf: Manifest[PROC_RES]): Future[RESPONSE] = {
+
+    val f : Future[PROC_RES] = ask(ref, cmd)(cmd.timeout).mapTo[PROC_RES]
+
+    val mpd: Future[RESPONSE] = f.map{x => result((None, Some(x)))}
+    mpd.recover {
+      case t : TimeoutException => result((Some(SexyTimeoutException(cmd.timeout)), None))
+      case th : Throwable => result((Some(th), None))
+    }
+  }
+
+  //possibly the WORST name ever.  Don't judge me.  'flatSend' because the above function is really just a fancy map function
+  //since you pass a func that goes from x->y and get a future[y], this is just the flatMap version of that.
+  def flatSend[PROC_RES,RESPONSE](cmd: BackgroundProcess[PROC_RES])(result: SendType[PROC_RES] => Future[RESPONSE])(implicit mf: Manifest[PROC_RES]): Future[RESPONSE] = {
```

Comment:
Yep...worst name ever.  I needed a send function which allowed me to flatMap on the result.